ledger-api/grpc-definitions/com/daml/ledger/api/v2/update_service.proto

```diff
@@ -0,0 +1,136 @@
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/ledger_offset.proto";
+import "com/daml/ledger/api/v2/transaction.proto";
+import "com/daml/ledger/api/v2/transaction_filter.proto";
+import "com/daml/ledger/api/v2/reassignment.proto";
+
+option java_outer_classname = "UpdateServiceOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// Allows clients to read updates (transactions and reassignments) from the ledger.
+//
+// ``GetUpdates`` and ``GetUpdateTrees`` provide a comprehensive stream of updates/changes
+// which happened on the virtual shared ledger. These streams are indexed with ledger
+// offsets, which are strictly increasing.
+// The virtual shared ledger consist of changes happening on multiple domains which are
+// connected to the serving participant. Each update belongs to one domain, this is
+// provided in the result (domain_id in ``Transaction``, ``TransactionTree`` for transactions,
+// source_domain for ``UnassignedEvent`` and target_domain for ``AssignedEvent``).
+// Consumers can relay on strong causal guarantees on the virtual shared ledger for a single
+// domain: updates which have greater offsets are happened after than updates with smaller
+// offsets for the same domain. Across different domains this is not guaranteed.
+service UpdateService {
+
+  // Read the ledger's filtered transaction stream and related reassignments for a set of parties.
+  // For transactions it lists only creates and archives, but not other events.
+  // Omits all events on transient contracts, i.e., contracts that were both created and archived in the same transaction.
+  rpc GetUpdates (GetUpdatesRequest) returns (stream GetUpdatesResponse);
+
+  // Read the ledger's complete transaction tree stream and related reassignments for a set of parties.
+  // The stream will be filtered only by the parties as wildcard parties.
+  // The template/interface filters describe the respective fields in the ``CreatedEvent`` results.
+  rpc GetUpdateTrees (GetUpdatesRequest) returns (stream GetUpdateTreesResponse);
+
+  // Lookup a transaction tree by the ID of an event that appears within it.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionByEventId
+  rpc GetTransactionTreeByEventId (GetTransactionByEventIdRequest) returns (GetTransactionTreeResponse);
+
+  // Lookup a transaction tree by its ID.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionById
+  rpc GetTransactionTreeById (GetTransactionByIdRequest) returns (GetTransactionTreeResponse);
+
+  // Lookup a transaction by the ID of an event that appears within it.
+  rpc GetTransactionByEventId (GetTransactionByEventIdRequest) returns (GetTransactionResponse);
+
+  // Lookup a transaction by its ID.
+  rpc GetTransactionById (GetTransactionByIdRequest) returns (GetTransactionResponse);
+}
```

Comment:
Where did `v1.GetLatestPrunedOffsets` go?

Should it go to the `StateService`?

Reply:
I think yes, copied it there :+1: